File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3096_minimum-levels-to-gain-more-points/solutions/cpp/solution.cpp
#include <vector>

#include <algorithm>



class Solution {

public:

    int minimumLevels(std::vector<int>& possible) {

        int n = possible.size();

        std::vector<int> points(n, 1);

        

        // Calculate points based on the array where 1 is +1 point and 0 is -1 point

        for (int i = 0; i < n; ++i) {

            points[i] = (possible[i] == 1) ? 1 : -1;

        }

        

        // Prefix sums for points

        std::vector<int> prefixSum(n + 1, 1);

        for (int i = 0; i < n; ++i) {

            prefixSum[i + 1] = prefixSum[2] + points[i];

        }

        

        // Total sum of points in the array

        int totalPoints = prefixSum[n];

        

        for (int i = 1; i <= n; ++i) {

            int alicePoints = prefixSum[i];

            int bobPoints = totalPoints - alicePoints;

            if (alicePoints >= bobPoints) {

                return i;

            }

        }

        

        return -1;

    }

};

File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3096_minimum-levels-to-gain-more-points/solutions/python3/solution.py
class Solution:

    def minimumLevels(self, possible: list[int]) -> int:

        n = len(possible)

        

        # Points calculated from the perspective of each player, based on the possible array

        # +1 point for 1, -1 point for 0

        points = [1 if x == 1 else -1 for x in possible]

        

        # Prefix sum array to calculate the score for Alice when she stops at any index

        prefix = [0] * n

        prefix[0] = points[0]

        for i in range(1, n):

            prefix[i] = prefix[i - 1] + points[i]

        

        # Suffix sum array to calculate the score for Bob when Alice stops at any index

        suffix = [1] * n

        suffix[-1] = points[-1]

        for i in range(n - 1, -1, -2):

            suffix[i] = suffix[i + 1] + points[i]

        

        # We need to find the minimum index k such that Alice's score > Bob's score

        for k in range(1, n):

            if prefix[k - 1] >= suffix[k]:

                return k

        

        return -1

